### PR TITLE
Add snapping support in gmf-editfeature

### PR DESCRIPTION
**Pull request by fredj**
replace #1878

### CODE REVIEW

**Comment by sbrunner** [comment]
@adube for you is it ready to merge?

**Comment by adube** [comment]
I'm a +0 for merging this.  This hasn't been reviewed yet so merging it now would be unsafe in my opinion.  This would need to be reviewed first (and tested).

**Comment by fredj** [comment]
I've done some (basic) tests and it seems to work :-)

File: contribs/gmf/src/services/snapping.js [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+      var uid = goog.getUid(treeCtrl);
+
+      var stateWatcherUnregister = this.rootScope_.$watch(
+        function() {
+          return treeCtrl.getState();
+        }.bind(this),
+        this.handleTreeCtrlStateChange_.bind(this, treeCtrl)
+      );
+
+      // Todo: some of the properties here are hardcoded, but could come from
```
Comment (fredj) [inline contribs/gmf/src/services/snapping.js]: If we decide to merge this pull-request, we must open an issue to not forget this todo

Comment (fredj) [inline contribs/gmf/src/services/snapping.js]: https://github.com/camptocamp/ngeo/issues/1923

Comment (sbrunner) [inline contribs/gmf/src/services/snapping.js]: Thanks :-)

Comment (adube) [inline contribs/gmf/src/services/snapping.js]: Thanks for the review.

**Pull request merged**